File: Typescript/timeConversion.ts
function timeConversion(s: string): string {
  // Write your code here
  const timeOfDay: string = s.slice(s.length - 2, s.length);
  const hours: number = Number(s.slice(0, 2));
  let convertedTime: string = "";
  
  
  if (timeOfDay == "AM"){
      hours == 12 ? convertedTime = "00" + s.slice(2, s.length - 2) : convertedTime = s.slice(0, s.length - 2);
  } else if (timeOfDay == "PM"){
      hours == 12 ? convertedTime = s.slice(0, s.length - 2) : convertedTime = String(12 + hours) + s.slice(2, s.length - 2);
  }
  
  
  return convertedTime;
}

const originalTime : string = "02:05:45AM";
const convertedTime : string = timeConversion(originalTime);
console.log(convertedTime);

File: Ruby/timeConversion.rb
def timeConversion(s)
  convertedTime = ""
  hours = s[0, 2]
  timeOfDay = s[8, 9]

  if timeOfDay.eql?("AM")
    hours == "12" ? convertedTime = "00".concat(s[2, 6]) : convertedTime = s[0, 8]
  elsif timeOfDay.eql?("PM")
    hours == "12" ? convertedTime = s[0, 8] : convertedTime = (12 + hours.to_i).to_s << s[2, 6]
  else 
    puts "Wrong input format"
  end
  
  return convertedTime
end

originalTime = "07:05:45PM";
convertedTime = timeConversion originalTime
puts convertedTime